shim_withdevtty_test.go

```diff
@@ -0,0 +1,20 @@
+// Copyright 2017 HyperHQ Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// +build linux darwin
+
+package main
+
+import (
+	"os"
+)
+
+func newTtyForTest() (*os.File, error) {
+	return os.OpenFile("/dev/tty", os.O_RDWR, os.ModeDevice)
```

Comment:
I don't think the test code can control this tty for testing.

pty.Open() in "github.com/kr/pty" provides you a ptsmaster which can be manipulated (maybe via some other package) for testing.

Reply:
The test only reads from it, which may be fine. I agree a ptsmaster is better here. Ans using `github.com/kr/pty` avoids OS specific implementation.